libs/build_processed_dataset.py

```diff
@@ -163,27 +202,50 @@ def get_usa_by_states_df(input_dir, intervention_type):
     )
 
     # basically the states_agg has full state names, the interventions have abbreviation so we need these to be merged
+    # inner merge to filter to only the 50 states+DC.  (left join to avoid missing data)
     states_abbrev = (
-        states_agg.merge(abbrev_df, left_index=True, right_on="state", how="left")
-        .merge(
-            # inner merge to filter to only the 50 states
-            interventions_df,
+        states_agg.merge(
+            abbrev_df,  # adds 'state', 'abbreviation'
+            left_index=True,
+            right_on="state",
+            how="left",
+        ).merge(
+            test_max_df,  # adds state, positive, negative
+            left_on="abbreviation",
+            right_on="abbreviation",
+            how="left",
+        ).merge(
+            interventions_df,  # add intervention columns
             left_on="abbreviation",
             right_on="state",
             how="inner",
-        )
-        .merge(projections_df, left_on="state_y", right_on="State", how="left")
-        .drop(["abbreviation", "state_y", "State"], axis=1)
+            suffixes=["", "_dropcol"],
+        ).drop(
+            "state_dropcol", axis=1
+        ).merge(
+            projections_df,  # add projection columns
+            left_on="abbreviation",
+            right_on="State",
+            how="left",
+        ).drop(["abbreviation", "State"], axis=1)
     )
 
-    states_remapped = states_abbrev.rename(columns=OUTPUT_COLUMN_REMAP_TO_RESULT_DATA)
+    STATE_COLS_REMAP = OUTPUT_COLUMN_REMAP_TO_RESULT_DATA
+    STATE_COLS_REMAP["positive"] = CUMULATIVE_POSITIVE_TESTS
+    STATE_COLS_REMAP["negative"] = CUMULATIVE_NEGATIVE_TESTS
+
+    states_remapped = states_abbrev.rename(columns=STATE_COLS_REMAP)
 
     states_final = pd.DataFrame(states_remapped, columns=RESULT_DATA_COLUMNS_STATES)
+    states_final[CUMULATIVE_POSITIVE_TESTS] = states_final[CUMULATIVE_POSITIVE_TESTS].fillna(0)
```

Comment:
hmm don't think we fill in zeros elsewhere? Not sure where we have landed on this.

Reply:
if we fill with zeros do we need the field to optional?